src/containers/CardList.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import FuncCard from "../components/FuncCard";
+import NavBar from "../components/NavBar";
+import variadic from 'variadic.js';
+
+class CardList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      subtitle: "Type numbers to test function",
+      funcNames: Object.keys(variadic),
```

Comment:
Maybe setState with this on:

 ```
componentDidMount() {
      this.setState({funcNames: Object.keys(variadic})
}
```

Reply:
I did this, but I don't see the benefit. I don't quite understand lifecycle yet.

Reply:
I would say one benefit of pulling something like this out of state directly is to put a default state in the event "variadic" goes down or isn't accessible which might result in sending a weird error (or more commonly, breaks other parts of the app that depend on that particular state).  It basically gives you more control over what is being presented to the user.